packages/aws-cdk-lib/aws-ec2/lib/instance.ts

```diff
@@ -589,12 +589,15 @@ export class Instance extends Resource implements IInstance {
     this.instancePublicDnsName = this.instance.attrPublicDnsName;
     this.instancePublicIp = this.instance.attrPublicIp;
 
+    if (props.initOptions?.timeout && props.resourceSignalTimeout) {
+      Annotations.of(this).addWarning('two fields set')
```

Comment:
You should use the `addWarningV2` method as the one you used it deprecated. Also give a better error mesaage.

Reply:
Addressed in new commit updates